thunder/core/transform_common.py

```diff
@@ -363,3 +367,101 @@ class PostOptimizationTransform(Transform, ABC):
     @abstractmethod
     def transform_trace(self, computation_trace: Trace, **kwargs):
         pass
+
+
+def functionalize_inplace_ops(computation_trace: Trace) -> list[Trace]:
+    """Functionalize in-place ops in ``computation_trace``.
+
+    This is a two-step function. The first step is to use the output of :func:`thunder.core.prims.copy_`'s
+    outputs as trace's outputs. The second step is to remove :func:`thunder.core.prims.copy_`
+    from :attr:`~thunder.core.symbol.Boundsymbol.subsymbols` if it's the last symbol.
+    """
+    from thunder.core import utils
+    from thunder.core.proxies import ProxyInterface
+    from thunder.core.symbol import VariableInterface
+    import thunder.torch
+
+    def is_inplace(bsym: BoundSymbol) -> bool:
+        if (isinstance(bsym.sym.id, str) and bsym.sym.id.endswith("_")) and bsym.subsymbols:
```

Comment:
We should probably add a tag for an operation being inplace instead of relying on parsing the id

Reply:
this is going to miss `copy_` in subsymbols.

i.e. thunder/torch/__init__.py::_batch_norm uses prims.copy_ in the middle of the op and won't be identified as inplace op by this